lib/index.js

```diff
@@ -179,6 +201,26 @@ FSTree.defaultIsEqual = function defaultIsEqual(entryA, entryB) {
   return equal;
 };
 
+FSTree.applyPatch = function(input, output, patch, _delegate) {
+  var delegate = assign({}, DEFAULT_DELEGATE, _delegate);
+  for (var i = 0; i < patch.length; i++) {
+    applyOperation(input, output, patch[i], delegate);
+  }
+};
+
+function applyOperation(input, output, operation, delegate) {
+  var method = operation[0];
+  var relativePath = operation[1];
+  var inputPath = path.join(input, relativePath);
+  var outputPath = path.join(output, relativePath);
+
+  if (delegate[method]) {
+    delegate[method](inputPath, outputPath, relativePath);
+  } else {
+    throw new Error('Unknown patch operation method: ' + method);
```

Comment:
The other possibility is if they pass in an invalid `patch` to the `applyPatch` function (unlikely though).

:+1: On the error message. Probably more useful in that form/similar.